regression/apps/mongoose/Makefile

```diff
@@ -1,47 +1,38 @@
 # SPDX-License-Identifier: MPL-2.0
 
-MAIN_MAKEFILE := $(firstword $(MAKEFILE_LIST))
-INCLUDE_MAKEFILE := $(lastword $(MAKEFILE_LIST))
-CUR_DIR := $(shell dirname $(realpath $(MAIN_MAKEFILE)))
+CUR_DIR := $(shell dirname $(realpath $(firstword $(MAKEFILE_LIST))))
 BUILD_DIR := $(CUR_DIR)/../../build/initramfs/regression/network
+MONGOOSE_DIR := $(CUR_DIR)/../../build/mongoose
+MONGOOSE_C := mongoose.c
+MONGOOSE_H := mongoose.h
+MONGOOSE_FILES := $(MONGOOSE_DIR)/$(MONGOOSE_C) $(MONGOOSE_DIR)/$(MONGOOSE_H)
+MONGOOSE_O := $(MONGOOSE_DIR)/mongoose.o
+SRC_SERVER := http_server.c
+SRC_CLIENT := http_client.c
 TARGET_SERVER := $(BUILD_DIR)/http_server
 TARGET_CLIENT := $(BUILD_DIR)/http_client
+BINS := $(TARGET_SERVER) $(TARGET_CLIENT)
 CC := cc
-CFLAGS := -W -Wall -Wextra -g -I. -DMG_HTTP_DIRLIST_TIME_FMT="%Y/%m/%d %H:%M:%S" -DMG_ENABLE_LINES=1 -DMG_ENABLE_IPV6=1 -DMG_ENABLE_SSI=1
-SRC_SERVER := http_server.c mongoose.c
-SRC_CLIENT := http_client.c mongoose.c
-DEP := mongoose.h
+CFLAGS := -W -Wall -Wextra -g -I. -I$(MONGOOSE_DIR) -DMG_ENABLE_LINES=1
+.PHONY: all clean
 
-.PHONY: all clean mongoose server client
+all: $(BINS) | $(BUILD_DIR)
```

Comment:
I think that it is `$(TARGET_SERVER)`/`$(TARGET_CLIENT)` that depends on `$(BUILD_DIR)`, instead of `all`.

So I suggest moving `| $(BUILD_DIR)` to the right place.